x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/deprecation_details_flyout.tsx

```diff
@@ -185,11 +196,30 @@ export const DeprecationDetailsFlyout = ({
         )}
 
         <EuiText>
-          {messages.map((m, i) => (
-            <p key={i} className="eui-textBreakWord">
-              {m}
-            </p>
-          ))}
+          {messages.map((currentMessage, i) => {
+            if (typeof currentMessage === 'object' && currentMessage.type === 'markdown') {
+              return (
+                <EuiMarkdownFormat
+                  key={i}
+                  className="eui-textBreakWord"
+                  textSize="relative"
+                  processingPluginList={processingPlugins}
+                  css={{ marginBlockEnd: lineHeightMedium }}
+                >
+                  {currentMessage.content}
+                </EuiMarkdownFormat>
+              );
+            }
+
+            const textContent =
+              typeof currentMessage === 'string' ? currentMessage : currentMessage.content;
```

Comment:
Is it possible that we might run into issues if `currentMessage.content` doesn't exist?

Reply:
I think we should be ok because the only valid types are `string` or `{ type: 'text' | 'markdown'; content: string; }`, and we account for each. Someone could do an unsafe cast when registering their deprecation, but that's true for most TS code.